ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+                       survival_time = NA, survival_time_unit=NA,  event_occured = NA)
+
+colData(GSE14671_SE) [["cancer_type"]] <- "Chronic myelogenous leukemia (CML)"
+colData(GSE14671_SE)[["treatment"]] <- "Imatinib"
+colData(GSE14671_SE)[["response"]] <- ifelse(grepl("NonResponder", colData(GSE14671_SE)$title), "NonResponder", "Responder")
+
+```
+
+```{r}
+#==== 3 ====
+GSE23554_SE <- run_all(raw_tar = "GSE23554_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE23554&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hgu133ahsensgcdf_24.0.0.tar.gz",
+                       delim = ".CEL.*" , GPL = "GPL96", cancer_type=NA,treatment = NA, survival_type = NA,
+                       event_occured = NA, response = "cisplatin response (complete response or incomplete response):ch1",
+                       survival_time_unit= NA, survival_time = "overall survival in days:ch1")
+
+colData(GSE23554_SE)[["cancer_type"]] <- "Ovarian Cancer"
+colData(GSE23554_SE)[["treatment"]] <- "Cisplatin"
+colData(GSE23554_SE)[["survival_type"]] <- "Overall_Survival"
+colData(GSE23554_SE)[["survival_time_unit"]] <- "day"
+
+```
+
+```{r}
+#==== 4 ====
+# R and NR based on the overal survival time.
+GSE14764_SE <- run_all(raw_tar = "GSE14764_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE14764&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hgu133ahsensgcdf_24.0.0.tar.gz",
+                       delim = ".cel.*" , GPL = "GPL96",cancer_type=NA, treatment = NA, response = "residual tumor:ch1", survival_type = NA,
+                       survival_time = "overall survival time:ch1", survival_time_unit= NA, event_occured = "overall survival event:ch1")
+
+colData(GSE14764_SE) [["cancer_type"]] <- "Ovarian Cancer"
+colData(GSE14764_SE) [["treatment"]] <- "Carboplatin+Paclitaxel (adjuvant)"
+colData(GSE14764_SE) [["survival_type"]] <- "Overall_Survival"
+colData(GSE14764_SE) [["survival_time_unit"]]  <- "month"
+
+```
+
+```{r}
+#==== 5 ====
+GSE20194_SE <- run_all(raw_tar = "GSE20194_RAW.tar" , 
+                       geo_path = "https://www.ncbi.nlm.nih.gov/geo/download/?acc=GSE20194&format=file" , 
+                       brain_path = "http://mbni.org/customcdf/24.0.0/ensg.download/hgu133ahsensgcdf_24.0.0.tar.gz",
+                       delim = "_.*" , GPL = "GPL96",cancer_type=NA, treatment = "treatment code:ch1" , response = "pcr_vs_rd:ch1",
+                       survival_type = NA, survival_time = NA, survival_time_unit= NA,  event_occured = NA)
+
+colData(GSE20194_SE)[["cancer_type"]] <- "Breast cancer"
+colData(GSE20194_SE)$treatment[is.na(colData(GSE20194_SE)$treatment) & colData(GSE20194_SE)$treatments.comments.ch1 == "Taxol x 12 FAC x 4"] <- "TFAC"
```

Comment:
Suggestions:

This not pleasant to read. It is often better to extract relevant data into variables with short but descriptive names to both save on typing and make code easier to read.

For example:
```
# extract the data
cData <- colData(GSE20194_SE)
has_na_treatment <- is.na(cData$treatment)
tx_comments <- cData$treatment.comments.ch1

# make the changes
cData[has_na_treatment & tx_comment == "Taxol x 12 FEC x4", ]$treatment <- "TFEC"
cData[has_na_treatment & tx_comment == "Taxol x 12  FEC x4", ]$treatment <- "TFEC"
cData[has_na_treatment & tx_comment == "FEC x 3 ( d/t N.C)Taxol x 12", ]$treatment <- "FECT"
cData[is.na(cData$treatment)]$treatment <- cData[is.na(cData$treatment)]$treatment.comments.chr

# update the object
colData(GSE20194_SE) <- cData
```

This also has the advantage that the actual SE colData is not modified until all the updates are complete, allowing you to check things are as you expect before modifying the actual SE.

NOTE: I have not tested this code, it is just an example of one way to improve readability.

Reply:
@NiktaFeizi